src/worker/circular-array.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * Default circular array size.
+ */
+export const DEFAULT_CIRCULAR_ARRAY_SIZE = 2000
+
+/** @inheritdoc */
```

Comment:
**suggestion:** We should define better docs here, cause I don't understand out of the box what a circular array could be and why it's needed :thinking:

Reply:
Documenting standard data structure like ring buffer or circular array build from built-in type is out of the scope.